src/Dafda/Configuration/ConsumerConfigurationBuilder.cs

```diff
@@ -97,7 +98,7 @@ public ConsumerConfigurationBuilder WithUnitOfWorkFactory(IHandlerUnitOfWorkFact
             return this;
         }
 
-        internal ConsumerConfigurationBuilder WithConsumerScopeFactory(Func<IServiceProvider, IConsumerScopeFactory> consumerScopeFactory)
+        internal ConsumerConfigurationBuilder WithConsumerScopeFactory(Func<IServiceProvider, IConsumerScopeFactory<MessageResult>> consumerScopeFactory)
```

Comment:
I'm sure you're aware, but this will introduce braking changes to the API - with roots in the change of how ConsumerScopeFactory works.